app/scripts/lib/fxa-client.js

```diff
@@ -98,14 +98,37 @@ function (_, FxaClient, $, xhr, p, Session, AuthErrors, Constants) {
     },
 
 
-    signIn: function (originalEmail, password, relier, options) {
+    /**
+     * Signin reasons are indicators to the backend that the signin
+     * is to complete a particular action. The backend can choose
+     * to perform actions, e.g., send emails, based on the reason.
+     */
+    SIGNIN_REASON: {
+      SIGN_IN: 'signin',
+      CHANGE_PASSWORD: 'change_password',
+      PASSWORD_RESET: 'password_reset',
+      ACCOUNT_UNLOCK: 'account_unlock'
+    },
+
+    signIn: function (originalEmail, password, reason, relier, options) {
```

Comment:
> This is growing a lot of arguments.

Indeed, I initially stuffed `reason` into the `options` block. We are inconsistent in our function signatures across the project. In some files, we use `options` blocks for all parameters, whereas in others (including this file) we use named parameters if a param is required and options if optional. Internally, I want `reason` to be required, even if it's optional on the backend.

> Can we use the options has for this new field, or is that for something else?

We could, but that's inconsistent within this file.